core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/joinHint/UseKeysOrIndexHint.kt

```diff
@@ -0,0 +1,89 @@
+package ch.ergon.dope.resolvable.clause.model.joinHint
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.clause.model.joinHint.UseKeysHintClass.Companion.UseKeysHint
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.formatListToQueryStringWithBrackets
+import ch.ergon.dope.resolvable.formatToQueryString
+import ch.ergon.dope.resolvable.fromable.IndexReference
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_FTS
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_GSI
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+
+interface UseKeysOrIndexHint : Resolvable
+
+class UseKeysHintClass private constructor(
+    private val useKeys: TypeExpression<out ValidType>,
+) : UseKeysOrIndexHint {
+    companion object {
+        @JvmName("singleUseKeysHintConstructor")
+        fun UseKeysHint(key: TypeExpression<StringType>) =
+            UseKeysHintClass(key)
+
+        @JvmName("multipleUseKeysHintConstructor")
+        fun UseKeysHint(keys: TypeExpression<ArrayType<StringType>>) =
+            UseKeysHintClass(keys)
+    }
+
+    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
+        val keysDopeQuery = useKeys.toDopeQuery(manager)
+        return DopeQuery(
+            queryString = "KEYS ${keysDopeQuery.queryString}",
+            parameters = keysDopeQuery.parameters,
+        )
+    }
+}
+
+fun useKeys(key: TypeExpression<StringType>) = UseKeysHint(key)
+
+fun useKeys(key: String) = useKeys(key.toDopeType())
```

Comment:
does _keys_ have to be plural if we just pass a single key?

Reply:
I mean it doesn't strictly _have_ to be plural, I just left it like that for consistency. Do you think it would be better to use the singular for these cases? Then I will change it in the `UseKeysClause` too.